pkg/applicationmanager/controllers/applicationbackup.go

```diff
@@ -757,34 +768,38 @@ func (a *ApplicationBackupController) backupVolumes(backup *stork_api.Applicatio
 		// Run any post exec rules once all volume backup is triggered
 		driverCombo := a.checkVolumeDriverCombination(backup.Status.Volumes)
 		// If the driver combination of volumes are all non-kdmp, call the post exec rule immediately
-		if driverCombo == nonKdmpDriverOnly {
-			// Let's kill the pre-exec rule pod here so that application specific
-			// data  stream freezing logic works. Certain app actually unleash the WRITE when session ends.
-			// For detail refer pb-3823
-			for _, channel := range terminationChannels {
-				logrus.Infof("Sending termination commands to kill pre-exec pod in non-kdmp driver path")
-				channel <- true
-			}
-			if backup.Spec.PostExecRule != "" {
-				log.ApplicationBackupLog(backup).Infof("Starting post-exec rule...")
-				err = a.runPostExecRule(backup)
-				if err != nil {
-					message := fmt.Sprintf("Error running PostExecRule: %v", err)
-					log.ApplicationBackupLog(backup).Errorf(message)
-					a.recorder.Event(backup,
-						v1.EventTypeWarning,
-						string(stork_api.ApplicationBackupStatusFailed),
-						message)
-					backup.Status.Stage = stork_api.ApplicationBackupStageFinal
-					backup.Status.FinishTimestamp = metav1.Now()
-					backup.Status.LastUpdateTimestamp = metav1.Now()
-					backup.Status.Status = stork_api.ApplicationBackupStatusFailed
-					backup.Status.Reason = message
-					err = a.client.Update(context.TODO(), backup)
+		if !a.execRulesCompleted[string(backup.UID)] {
+			if driverCombo == nonKdmpDriverOnly {
+				// Let's kill the pre-exec rule pod here so that application specific
+				// data  stream freezing logic works. Certain app actually unleash the WRITE when session ends.
+				// For detail refer pb-3823
+				// Todo: get-rid of passing terminationChannel as argument, use the method reciever structure to access via backup UID.
+				for _, channel := range terminationChannels {
+					logrus.Infof("Sending termination commands to kill pre-exec pod in non-kdmp driver path")
+					channel <- true
+				}
+				if backup.Spec.PostExecRule != "" {
+					log.ApplicationBackupLog(backup).Infof("Starting post-exec rule...")
+					err = a.runPostExecRule(backup)
+					a.execRulesCompleted[string(backup.UID)] = true
```

Comment:
We also call `runPostExecRule()` for kdmp and mixeddriver as here also first time postexec rule can take longer time and we end up calling this again.

Reply:
We will not hit this path for kdmp, there is only one case wherein we are returning the code but that is sounds like a no-op so far. Hence put an info and a warning message so that if in case that path hit we will know. Currently at this time not preferable to change a working code. Hence added additional logs